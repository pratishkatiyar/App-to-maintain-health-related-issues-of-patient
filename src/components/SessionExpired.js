import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { Redirect } from 'react-router-dom';
import Navbar from './NavBar';

export default class SessionExpired extends Component {
    render() {
        const itemStr = localStorage.getItem("token")
        if(itemStr){
            const item = JSON.parse(itemStr)
            const now = new Date()
            if (now.getTime() < item.expiry) {
                return <Redirect to="/Dashboard"/>
            }else{
                localStorage.removeItem("token")
            }
        }        
        return (
            <div data-testid='sessionExpired-1'>
                <Navbar/>
                <h1>Session Expired.</h1>
                <Link to = '/'><button className="button-19">Login Again</button></Link>
            </div>
        )
    }
}
