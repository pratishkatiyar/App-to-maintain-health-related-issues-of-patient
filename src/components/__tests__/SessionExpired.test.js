import { render, screen, cleanup } from '@testing-library/react'
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import SessionExpired from '../SessionExpired'
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

afterEach(() => {
    cleanup();
})

test('should render SessionExpired component', () =>{
    render(<BrowserRouter><SessionExpired/></BrowserRouter>);
    const sessionExpiredElement = screen.getByTestId('sessionExpired-1');
    expect(sessionExpiredElement).toBeInTheDocument();
    expect(sessionExpiredElement).toHaveTextContent('Session Expired.');
})

test('matches snapshot', ()=> {
    const tree = renderer.create(<BrowserRouter><SessionExpired/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})