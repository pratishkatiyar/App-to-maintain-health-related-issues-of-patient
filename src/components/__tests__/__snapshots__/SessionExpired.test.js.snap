// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="sessionExpired-1"
>
  <div
    data-testid="navbar-1"
  >
    <nav>
      <div
        className="logo"
      >
        <p>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </p>
      </div>
      <div
        className="logo"
      >
        Patient Registration App
      </div>
      <div />
    </nav>
  </div>
  <h1>
    Session Expired.
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="button-19"
    >
      Login Again
    </button>
  </a>
</div>
`;
